### ISSUE TITLE
Queue is cleared after first job finishes

### ISSUE THREAD

**Issue by gabriels1234** [dsoftwareinc/django-tasks-scheduler#50]
**Describe the bug**
It was hard to track down, but basically, I have a Cron Job that schedules 26 jobs. The first job finishes (as per the worker's logs) and then it stops working. and upon refreshing the RQ Queues, the queue goes from 25 queued jobs to 0.
This also happen when enqueuing via code

**To Reproduce**
Add some code that does:
```
queue = get_queue()
for i in range(20):
  queue.enqueue(some_func, job_id=f"job_{i}", args=(a,b))
```

**Expected behavior**
When queuing more than one job, all the jobs must be performed.

**Screenshots**

**Desktop (please complete the following information):**
 - python 3.11
 - django 4.2.6
 - requirements.txt? django-task-scheduler==1.2.3

**Additional context**
My project relies on being able to queue jobs programatically (usually more than one at a time). I'm still investigating the root cause, but I've tried using the DjangoQueue and the normal Queue and the results are the same.

**Comment by cunla** [comment]
Hi, if I understand correctly, you have a corn-task executing a method that enqueues 26 jobs and after the first out of 26 jobs is terminated, the other jobs are not in the queue anymore?

- How many workers do you have?
- Are both the cron-job and the enqueued jobs in the same queue? or the cron-job in queue1 and the jobs it enqueues are in queue2?
- After the cron-job executes, ie, you enqueued 26 jobs 
  - what is the status of the queues? how many jobs are finished? how many are queued? how many failed? etc.
- After the first job executes, you mentioned there are no longer queued jobs
  - Do they appear in finished/failed/...?
  - What is the successful jobs/failed jobs counters of the worker show?

**Comment by danielm-cfa** [comment]
I have written a test for the scenario you described.. Looks like it executes all the 20 queued job. 
Let me know what I am missing..

https://github.com/dsoftwareinc/django-tasks-scheduler/commit/13571f138f56a7c878840b588d10e0354ba52712

**Comment by cunla** [comment]
I have written a test to address this. https://github.com/dsoftwareinc/django-tasks-scheduler/commit/13571f138f56a7c878840b588d10e0354ba52712

Looks like it works.. let me know what I am missing

**Comment by gabriels1234** [comment]
Thanks!
I've been trying many ways, still cannot make it work reliably.
Since our current code was originally done for the original django-rq-scheduler, there might be things that we are not seeing but that are different. I tried trying to enqueue directly to the Queue (rq's original) object, but also did not help.

At the moment we have no alternative other than reverting to django-rq-scheduler v2023.4.0 which is the latest before the breakup with django-rq. I will try to not use any django-rq feature (like get_queue() ) so I could test in the future.

Thank you so much with your help, if you think there's a place I should look into (what could make that the dashboard to go from 25 queued jobs to 0 without any logging in the worker) that would be great.
Have a Nice weekend!

**Comment by cunla** [comment]
So the jobs must go "somewhere" unless redis is manipulated externally.

When jobs are queued check their IDs (using UI/code), 
and then after they disappear (ie, not in any status) - run `Job.fetch(//ID//, connection=queue.connection)`.

=> If the job is totally gone - ie, `Job.fetch` returns `None` -
it means the job was removed completely from Redis, most likely within the logic of the first job?!

**Comment by gabriels1234** [comment]
UPDATE: 
I've found that redis is getting surprise "FLUSHDB" commands.
Consider redis is also used for general caching, and for the popular Django-Constance.
in the past, the project was using django-redis-cache for the cache backend.
However, as of Django 4, there's a native Redis cache backend.
Latest versions from django-tasks-scheduler conflict with django-redis-cache, which I understand.
So, as of now I've seen success by changing the Database (number) for the RQ queues (SCHEDULER_QUEUES)

please consider `scheduler.queues's `

```
    return redis_cls(
        host=config['HOST'],
        port=config['PORT'],
        db=config.get('DB', 0),  # This need to be set in settings.py, and a default could be added to this repo.
        .....
```

Will update you further later.

**Comment by cunla** [comment]
Use a different redis DB for cache and for pubsub (workers/tasks).
A redis instance has 16 DBs, use 0 for cache, 1 for pubsub, etc.

**Comment by gabriels1234** [comment]
Yes, I did and it worked. But it was working before, up to 2023.4.0 (or even more) smoothly.

**Comment by cunla** [comment]
Well, django-tasks-scheduler (or django-rq-scheduler) doesn't have direct redis commands.
So, I don't see how it can work with earlier versions if all other versions are equal.
The flushdb is coming either from the cache mechanism or from a newer version of rq.

**Comment by cunla** [comment]
Any luck replicating?
If not, can you close this?

**Comment by gabriels1234** [comment]
Hi, Sorry for not getting back earlier.
This issue was caused by the change from the redis_backend (don't remember the name) to the new Django native redis backend.

Basically, if redis is being used for something else (Constance in my case), and that makes a FLUSHDB call, then, of course, all the jobs would be deleted.

Solution was: use different DB-Number (by default from 0 to 15). I put rq on DB-0 and the other on DB-15. that way, the FLUSHDB does not affect rq.

Leaving here to help others troubleshoot.
